File: life-os-ui/src/app/models/routine.ts
export interface Routine {
  id: string;
  name: string;
  startTime: string; // ISO time format "HH:mm:ss"
  endTime: string;
  daysOfWeek: string; // "MONDAY,TUESDAY..."
  isBlocking: boolean;
}

export interface RoutineRequest {
  name: string;
  startTime: string;
  endTime: string;
  daysOfWeek: string[];
  isBlocking: boolean;
}

File: life-os-backend/src/main/java/com/lifeos/model/Routine.java
package com.lifeos.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalTime;
import java.util.UUID;

@Entity
@Table(name = "routines")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Routine {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private LocalTime startTime;

    @Column(nullable = false)
    private LocalTime endTime;

    /**
     * Stored as comma-separated days: "MONDAY,TUESDAY,WEDNESDAY"
     */
    @Column(nullable = false)
    private String daysOfWeek;

    @Column(nullable = false)
    private boolean isBlocking = true;
}
